pymedphys/_dicom/rtplan/helpers.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (C) 2018 Cancer Care Associates
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+"""Some helper utility functions for accessing DICOM RT Plan info.
+"""
+
+import pandas as pd
+
+import pydicom
+
+from pymedphys._mosaiq.constants import TOLERANCE_TYPES
+
+
+def get_all_dicom_treatment_info(dicomFile):
+    dicom = pydicom.dcmread(dicomFile)
+    dicomBeam = []
+    dicomData = []
+    prescriptionDescription = dicom.PrescriptionDescription.split("\\")
+    for fraction in dicom.FractionGroupSequence:
+        for beam in fraction.ReferencedBeamSequence:
+            dicomBeam = []
+            bn = (
+                beam.ReferencedBeamNumber
+            )  # pull beam reference number for simplification
+            doseRef = fraction.ReferencedDoseReferenceSequence[
+                0
+            ].ReferencedDoseReferenceNumber  # pull dose reference number for simplification
+            fn = fraction.FractionGroupNumber
+            dicomBeam.append(dicom.RTPlanName)  # add prescription plan name
+            dicomBeam.append(dicom.PatientID)  # add patient ID
```

Comment:
The way this function is designed it is quite tricky to get a handle on the alignment between column name defined below, and each item as defined by appending to a list in order.

Maybe, instead of creating a list at the beginning, start with an empty pandas dataframe and grow it, column by column, as described in the following tutorial:

https://www.geeksforgeeks.org/adding-new-column-to-existing-dataframe-in-pandas/

It'll likely run slower (minutely), but it will be far easier to read, and significantly less likely to make a mistake in data alignment.